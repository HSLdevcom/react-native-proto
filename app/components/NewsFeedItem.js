/**
 * NewsFeedItem component
 * @flow
 */

import React, {Component} from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import Immutable from 'immutable';
import moment from 'moment';
import colors from '../colors';

const styles = StyleSheet.create({
    text: {
        color: colors.brandColor,
        marginBottom: 15,
    },
});

class NewsFeedItem extends Component { // eslint-disable-line react/prefer-stateless-function
    handleShowSingle = () => {
        const {data, showSingle} = this.props;
        showSingle(data.get('nid'));
    }
    renderItem = ({item}) =>
        <Text key={item.nid} style={styles.text}>{`${moment.unix(item.created).format('DD.MM.YYYY')}\n${item.title}`}</Text>

    render() {
        const {data} = this.props;
        return (
            <TouchableOpacity key={data.get('nid')} onPress={this.handleShowSingle}>
                <Text style={styles.text}>
                    {`${moment.unix(data.get('created')).format('DD.MM.YYYY')}\n${data.get('title')}`}
                </Text>
            </TouchableOpacity>
            // We can use FlatList with react-native 0.43
            // newsList = (
            //     <FlatList
            //         data={data}
            //         renderItem={this.renderItem}
            //     />
            // );
        );
    }
}

NewsFeedItem.propTypes = {
    data: React.PropTypes.instanceOf(Immutable.Map).isRequired,
    showSingle: React.PropTypes.func.isRequired,
};

export default NewsFeedItem;
