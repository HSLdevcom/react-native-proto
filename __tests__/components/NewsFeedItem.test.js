// https://facebook.github.io/jest/blog/2016/07/27/jest-14.html
import 'react-native';
import React from 'react';
import {fromJS} from 'immutable';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import NewsFeedItem from '../../app/components/NewsFeedItem';

it('renders correctly', () => {
    const tree = renderer.create(
        <NewsFeedItem showSingle={jest.fn()} data={fromJS({})} />).toJSON();
    expect(tree).toMatchSnapshot();
});
