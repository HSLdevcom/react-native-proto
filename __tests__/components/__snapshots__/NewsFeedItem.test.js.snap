// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(0, 122, 201)",
        "marginBottom": 15,
      }
    }
  >
    Invalid date
    undefined
  </Text>
</View>
`;
